### PR TITLE
feat: option for max operating duration

### PR DESCRIPTION
**Pull request by ssweriduk**
### Description
Enables the ability to run this program for a specified period of time. This is especially useful when defining windows of operation.
For example, I want this program to only search for cards during the hours of 9AM -> 11PM as I don't want to be woken up in the middle of the night for card alerts. Setting a cron schedule to run daily at 9AM and a `MAX_RUNTIME` env var of `14400` achieves this. 
If I won't be actionable on the alerts this program sends out, why have it running. 
This is also useful for anybody looking to run this in a containerized environment in the cloud, and don't want to incur extra server costs during scheduled downtimes 

### Testing

Tested locally. Don't see any unit tests in this project, but happy to add some if needed!

### CODE REVIEW

File: src/store/lookup.ts [hunk 1]
```diff
@@ -218,7 +218,7 @@ async function lookupCardInStock(store: Store, page: Page, link: Link) {
 }
 
 export async function tryLookupAndLoop(browser: Browser, store: Store) {
-	if (!browser.isConnected()) {
+	if (!browser || !browser.isConnected()) {
```
Comment (ssweriduk) [inline src/store/lookup.ts:221]: `stopAndExit` sets `browser` to undefined. Adding this check here to gracefully exit this loop.

File: src/index.ts [hunk 2]
```diff
@@ -69,6 +69,15 @@ async function main() {
...
 
 	await startAPIServer();
+
+	if (config.maxRunTime) {
+		setTimeout(async () => {
+			logger.info(
+				`We've been searching for over ${config.maxRunTime} seconds as configured by MAX_RUNTIME. Time to call it quits for now. Gracefully stopping program.`
+			);
+			await stopAndExit();
+		}, config.maxRunTime);
```
Comment (jef) [inline src/index.ts:79]: This is in milliseconds, so you will have to convert.

**Review by jef — COMMENTED** [review]
I could see how someone would want this feature, but then I could also see how this could be accomplished outside of the app. Perhaps even cause confusion (in terms of automation).

I'm just a little nervous in pulling this in, cause I don't know how much benefit it will add and perhaps cause confusion. But the change is so small, it shouldn't be a problem. So I'm not sure.

What do you think?

**Comment by SesioN** [comment]
@ssweriduk 

I solved your issue in a different way, which should be better applicable.

https://github.com/jef/streetmerchant/issues/629

However I never got to refine this and make a PR. 

If you want, you could transform this into your own PR. I linked my dirty solution in the topic.

File: src/index.ts (hunk 2 above)
Comment (ssweriduk) [reply src/index.ts:79]: Doh! Thanks for catching.

**Comment by ssweriduk** [comment]
@SesioN sorry for the delay! Holidays etc etc. Lemme take a look at that PR! Thanks for looking at this!

**Comment by ssweriduk** [comment]
@jef Thanks for reviewing! And I agree, it could be confusing, I was struggling to come up with an env var name for this. I don't believe that AWS ECS has a way of sending `SIGTERM` to the containers on a schedule without setting up some sort of CloudWatch/Lambda integration, which I was trying to avoid. Open to other solutions!

File: src/store/lookup.ts (hunk 1 above)
Comment (jef) [reply src/store/lookup.ts:221]: Oh! I think this could help #1174. Not sure though, haven't looked into in depth.

**Review by jef — CHANGES_REQUESTED** [review]
After we fix that up, I think we can pull this in.

Thanks!

**Pull request closed without merging**